src/app/rpmostree-compose-builtin-tree.cxx

```diff
@@ -948,13 +939,36 @@ impl_install_tree (RpmOstreeTreeComposeContext *self,
   if (!inject_advisories (self, cancellable, error))
     return FALSE;
 
+  /* Insert our input hash */
+  g_hash_table_replace (self->metadata, g_strdup ("rpmostree.inputhash"),
+                        g_variant_ref_sink (g_variant_new_string (new_inputhash)));
+
   /* Destroy this now so the libdnf stack won't have any references
    * into the filesystem before we manipulate it.
    */
   g_clear_object (&self->corectx);
 
   if (g_strcmp0 (g_getenv ("RPM_OSTREE_BREAK"), "post-yum") == 0)
-    return FALSE;
+    return glnx_throw (error, "Aborting early due to RPM_OSTREE_BREAK flag.");
+
+  *out_changed = TRUE;
+  return TRUE;
+}
+
+
+static gboolean
+impl_install_tree_post (RpmOstreeTreeComposeContext *self,
+                        GCancellable    *cancellable,
+                        GError         **error)
+{
+  /* Pick the next version label from metadata */
+  gpointer vp = g_hash_table_lookup (self->metadata, OSTREE_COMMIT_META_KEY_VERSION);
+  if (vp == NULL)
```

Comment:
My bad, I was too harsh here, this is one of the places where we use empty strings to represent `Option<String>`.